client/my-sites/domains/domain-search/index.jsx

```diff
@@ -158,7 +162,23 @@ class DomainSearch extends Component {
 				<span>
 					<div className="domain-search__content">
 						<PlansNavigation cart={ this.props.cart } path={ this.props.context.path } />
-
+						{ hasCustomDomain( selectedSite ) && (
+							<Notice
+								status="is-warning"
+								showDismiss={ false }
+								text={
+									translate( "You're purchasing an additional domain for %(title)s.", {
+										args: { title: selectedSiteTitle },
+									} ) +
+									' ' +
+									translate( 'Would you like to create a new site instead?' )
+								}
+							>
+								<NoticeAction href="/jetpack/new?ref=calypso-selector">
```

Comment:
This should send users to `/start` (NUX) rather than a `/jetpack/` flow.

Reply:
Updated! My only concern with that URL though is that, unlike the previous one, users lose access to the Back button. It's also the link people are redirected to when clicking "Add New Site". 

<img width="165" alt="Screenshot 2019-07-30 at 13 29 24" src="https://user-images.githubusercontent.com/43215253/62129239-24af8300-b2ce-11e9-8880-ef433e1c2230.png">

Reply:
> My only concern with that URL though is that, unlike the previous one, users lose access to the Back button. It's also the link people are redirected to when clicking "Add New Site".

I see. Not sure if we have an easy way to do that with `/start`.

Reply:
It looks like the "Back to My Sites" link from `/start` was originally added under the assumption that there wouldn't be a case like this. From @sixhours in #29952:

> Taking a step back and looking at this again, **we're assuming the user has clicked on My Sites, switched sites, and opted to add a new site**. Therefore, sending them back to /sites makes sense to me, and we can use the (already familiar) language "My Sites", which is what they'll see at that location.

<img width="188" alt="Screenshot 2019-07-30 at 14 37 14" src="https://user-images.githubusercontent.com/43215253/62133906-8a543d00-b2d7-11e9-8300-48fa0c3efa83.png">

The solutions probably are:

1. Just don't offer a Back link for this flow
2. Change "Back to My Sites" into an ordinary "Back" link
3. Switch this link back to `/jetpack/new?ref=calypso-selector` again

I was inclined to go with two, but even that's a bit tricky to implement as `/start` redirects a lot. Maybe `/jetpack/new?ref=calypso-selector` is the better link to go with after all?

Reply:
At this point I'm inclined to see what the @Automattic/zelda thinks.

Reply:
> Maybe /jetpack/new?ref=calypso-selector is the better link to go with after all?

I'm inclined to agree with this, only because it matches the logged-in behaviour of **+ ADD NEW SITE** (in the side bar) and it keeps things simple.

The link goes here.

<img width="600" alt="Screen Shot 2019-07-31 at 10 28 06 am" src="https://user-images.githubusercontent.com/6458278/62174610-f12c3180-b37d-11e9-808a-b324a8508ad8.png">

It might not be ideal as it introduces the notion of 'Jetpack' to the user, which might be confusing, but @aidvu is right: the signup flow messes with the route history. It shunts the user along to the most appropriate step, which is, in most cases `/start/site-type`. 

In this situation, using the  `getPreviousRoute` selector, which might have been a possible solution to the back button url dilemma, will return `/start` and not `/domains/add/{site}`

**TL;DR**  The path of less resistance is to use `/jetpack/new?ref=calypso-selector`

One more thing, in my opinion, it should be `'Create a new site'`, with the indefinite article.

Reply:
I also ran into that issue when trying to use `getPreviousRoute`, updated the URL to `/jetpack/new?ref=calypso-selector` now. Also added the indefinite article, thanks @ramonjd! :)